core/src/dird/ua_purge.cc

```diff
@@ -262,47 +278,93 @@ static bool PurgeFilesFromClient(UaContext* ua, ClientResource* client)
   return true;
 }
 
+static bool PurgeJobsFromPool(UaContext* ua, PoolDbRecord* pool_record)
+{
+  std::vector<char> jobstatuslist;
+  if (!GetUserJobStatusSelection(ua, jobstatuslist)) {
+    ua->ErrorMsg(_("invalid jobstatus parameter\n"));
+    return false;
+  }
+
+  std::string select_jobs_from_pool
+      = "SELECT DISTINCT JobMedia.JobId FROM JobMedia, Job "
+        "WHERE Job.JobId = JobMedia.JobId "
+        "AND Jobmedia.MediaId IN "
+        "(SELECT MediaId FROM Media WHERE PoolId="
+        + std::to_string(pool_record->PoolId) + ") ";
+
+  if (!jobstatuslist.empty()) {
+    std::string jobStatuses
+        = CreateDelimitedStringForSqlQueries(jobstatuslist, ',');
+    select_jobs_from_pool += "AND Job.JobStatus IN (" + jobStatuses + ") ";
+  }
+  select_jobs_from_pool.append("ORDER BY JobId");
+
+
+  std::vector<JobId_t> delete_list;
+  ua->db->SqlQuery(select_jobs_from_pool.c_str(), JobDeleteHandler,
+                   &delete_list);
+
+  if (delete_list.size() > 0) { PurgeJobListFromCatalog(ua, delete_list); }
+
```

Comment:
When  purging all jobs from a volume, the volume is still marked as full. When purging the last job from a volume, the volume will be marked as purged afterwards.